File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int s1 = 1;
        int s2 = 1;
        for (int i = 2; i < n + 1; ++i) {
            int temp = s2;
            s2 = s2 + s1;
            s1 = temp;
        }
        
        return s2;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        memo = {}

        def count(i):
            if i == 0:
                return 1

            if i == 1:
                return 1

            if i in memo:
                return memo[i]

            memo[i] = count(i - 1) + count(i - 2)
            return memo[i]

        return count(n)
                